File: Horse-racing Duals.py
import sys
import math

horses = []
lowest = 100000

n = int(input())
for i in range(n):
    pi = int(input())
    horses.append(pi)
horses.sort(reverse=True)
lasthorse = horses[-1] 


def getdist(item1, item2):
    """ Returns the distance between two values... """
    return item1 - item2


for horse in range(len(horses) - 1):
    horseOne = horses[horse]
    horsetwo = horses[horse + 1]
    if getdist(horseOne, horsetwo) < lowest:
        lowest = getdist(horseOne, horsetwo)
    
   
print(lowest)

File: Horse-racing Duals.java
import java.util.*;
import java.io.*;
import java.math.*;

class Solution {

    public Integer getDist(int item1, int item2){
        return (item1 - item2);
    }
         
    public static void main(String args[]) {
        
        Solution program = new Solution();
        int lowest = 10000;
        ArrayList<Integer> horses = new ArrayList<>();
        Scanner in = new Scanner(System.in);
        int N = in.nextInt();
        for (int i = 0; i < N; i++) {
            int pi = in.nextInt();
            horses.add(pi);
    }

        Collections.sort(horses, Collections.reverseOrder());
        
        for(int horse = 0; horse < horses.size() - 1; horse++){
            int horseOne = horses.get(horse);
            int horseTwo = horses.get(horse + 1);
            if (program.getDist(horseOne, horseTwo) < lowest){
                    lowest = program.getDist(horseOne, horseTwo);
                }
            }
        System.out.println(lowest);
    }
}

